### PR TITLE
Remove prop-types from graphie.tsx

### PR DESCRIPTION
**Pull request by jeremywiebe**
## Summary:

Switches to true TypeScript types for `graphie.tsx`.

Screenshot 2023-04-24 at 12 46 37 PM

Issue: "none"

## Test plan:

Viewed the Graphie stories in storybook.

### CODE REVIEW

File: packages/perseus/src/components/graphie.tsx [hunk 1]
```diff
@@ -131,8 +156,7 @@ class Graphie extends React.Component<Props> {
     _setupGraphie: () => void = () => {
         this._removeMovables();
 
-        // eslint-disable-next-line react/no-string-refs
-        const graphieDiv = ReactDOM.findDOMNode(this.refs.graphieDiv);
+        const graphieDiv = ReactDOM.findDOMNode(this.graphieDivRef.current);
```
Comment (jeremywiebe) [inline packages/perseus/src/components/graphie.tsx:159]: I also switched out the string-ref for a true `React.createRef()` ref.

File: packages/perseus/src/components/svg-image.tsx [hunk 2]
```diff
@@ -240,10 +241,7 @@ class SvgImage extends React.Component<Props, State> {
             labelDataIsLocalized: false,
             labels: [],
             labelsRendered: {},
-            range: [
-                [0, 0],
-                [0, 0],
-            ],
+            range: [[0, 0], [0, 0] as [number, number]],
```
Comment (handeyeco) [inline packages/perseus/src/components/svg-image.tsx:244]: Just curious about this one. Was TS interpreting these as having to be zeros? Like `[0, 0]`? All these things called `range` is throwing me off, but I'm wondering if we could have a shared type for these.

File: packages/perseus/src/components/svg-image.tsx [hunk 3]
```diff
@@ -719,14 +717,12 @@ class SvgImage extends React.Component<Props, State> {
...
 
-            const scale = [40 * this.props.scale, 40 * this.props.scale];
-
             graphie = (
                 <Graphie
                     // eslint-disable-next-line react/no-string-refs
                     ref="graphie"
                     box={box}
-                    scale={scale}
+                    scale={[40 * this.props.scale, 40 * this.props.scale]}
```
Comment (handeyeco) [inline packages/perseus/src/components/svg-image.tsx:725]: Do we know what this magic number is?

File: packages/perseus/src/perseus-types.ts [hunk 4]
```diff
@@ -3,6 +3,9 @@
 
 import type {Coord} from "./interactive2/types";
 
+export type Range = [min: number, max: number];
+export type Size = [width: number, height: number];
```
Comment (handeyeco) [inline packages/perseus/src/perseus-types.ts:7]: I'm not familar with this, but I'm interpreting it as `[number, number]`, but you're naming the indices?

File: packages/perseus/src/perseus-types.ts [hunk 5]
```diff
@@ -1162,7 +1165,7 @@ export type PerseusInteractionGraph = {
     // How many ticks to show on the ruler.  e.g. 1, 2, 4, 8, 10, 16
     rulerTicks?: number;
     // This controls the number (and position) of the tick marks for the X and Y axis. e.g. [1, 1]
-    tickStep: ReadonlyArray<number>;
+    tickStep: [x: number, y: number];
```
Comment (handeyeco) [inline packages/perseus/src/perseus-types.ts:1168]: Is this syntax `Readonly`? Or are we intentially not making these `Readonly`?

File: packages/perseus/src/widgets/simulator.tsx [hunk 6]
```diff
@@ -231,9 +231,7 @@ class Histogram extends React.Component<any, any> {
             range: range,
             data: data,
             scale: [
-                // @ts-expect-error [FEI-5003] - TS2345 - Argument of type 'any[]' is not assignable to parameter of type 'Coordinates'.
```
Comment (handeyeco) [inline packages/perseus/src/widgets/simulator.tsx:234]: Nice

**Review by handeyeco — APPROVED** [review]
I'm confident you know what you're going, just want to make sure we're intentionally making things mutable vs read-only.

File: packages/perseus/src/components/graphie.tsx [hunk 7]
```diff
@@ -1,45 +1,67 @@
...
+    gridStep?: [number, number];
+    step?: [number, number];
+    scale?: [number, number];
+
+    isMobile?: boolean;
+    responsive?: boolean;
+
+    children?: React.ReactNode;
+
+    options: any;
```
[the remark this answers is not in the document]
Comment (jeremywiebe) [reply packages/perseus/src/components/graphie.tsx:37]: Done! https://khanacademy.atlassian.net/browse/LC-772

File: packages/perseus/src/components/svg-image.tsx (hunk 2 above)
Comment (jeremywiebe) [reply packages/perseus/src/components/svg-image.tsx]: Those literals are inferred as `[]number` (unbounded arrays).  So the `as` makes it explicit that they're 2-item tuples.

File: packages/perseus/src/components/svg-image.tsx (hunk 3 above)
Comment (jeremywiebe) [reply packages/perseus/src/components/svg-image.tsx:726]: No idea. :(  I'll try to figure it out.

File: packages/perseus/src/perseus-types.ts (hunk 4 above)
Comment (jeremywiebe) [reply packages/perseus/src/perseus-types.ts:7]: It's a helper syntax to give more meaning to the tuple.  You're exactly correct, they're just labels/names, but I don't think you can even reference the tuple items using them, so they're strictly decorative. They are picked up by the editor which gives the hint as well. 

image

**Pull request merged**